### PR TITLE
Add JSHinting to travis.

### PR DESCRIPTION
**Pull request by koddsson**
jshint is now run on travis!

### CODE REVIEW

**Comment by koddsson** [comment]
Related to #103

File: .jshintrc [hunk 1]
```diff
@@ -0,0 +1,3 @@
+{
+    "esnext": true
+}
```
Comment (koddsson) [inline .jshintrc]: I don't know what we'd want here but we can add to this as we go along. If you want to add something specific, feel free!

**Comment by koddsson** [comment]
Not sure why this isn't building on travis :<

**Comment by koddsson** [comment]
Don't merge this yet. It doesn't seem to be reporting jshint errors from travis still.

File: .travis.yml [hunk 2]
```diff
@@ -1,16 +1,17 @@
 language: node_js
+before_script:
+- jshint .
 node_js:
 - '0.11'
+- '0.10'
```
Comment (kristjanmik) [inline .travis.yml]: We are only testing for 0.11 for now

File: .travis.yml [hunk 3]
```diff
@@ -1,16 +1,17 @@
 language: node_js
+before_script:
```
Comment (kristjanmik) [inline .travis.yml]: This might be running before the jshint is installed, you could try doing

after_script:
  - "jshint --config=.jshintconfig ."

Comment (kristjanmik) [inline .travis.yml]: Or just add it to the beginning of after_success

File: .travis.yml (hunk 2 above)
Comment (koddsson) [inline .travis.yml]: It keeps creeping in! I'll remove it!

**Comment by koddsson** [comment]
Ok this is weird. Because it should fail because we've got jshint errors in our code right now. I'll investigate further.

**Comment by koddsson** [comment]
Ah no! I got it, all the errors when running locally are in `./dist/` so they don't exist on the travis box when it's at that stage yet (since nothing has been built).

tl;dr this works as expected and is ready for merging given that it passes code-review.

cc/ @MiniGod @benediktvaldez @kristjanmik

File: .travis.yml [hunk 4]
```diff
@@ -3,14 +3,12 @@ node_js:
 - '0.11'
 before_install:
 - npm install -g npm@latest
+- npm install -g jshint
 - openssl aes-256-cbc -K $encrypted_893d0cdd819b_key -iv $encrypted_893d0cdd819b_iv -in travisapis.enc -out travisapis -d
 script: npm test
-branches:
-only:
-- master
```
Comment (kristjanmik) [inline .travis.yml]: This will cause the server to rebuild on every commit. I guess it is not related to this though since I will take a look at the deployment pipeline tomorrow
